getstarted.md

```diff
@@ -3,72 +3,203 @@ layout: page
+
+If your client entry point looks like this:
+
+```js
+import React from 'react';
+import { render } from 'react-dom';
+import RootContainer from './containers/rootContainer.js';
+
+render(<RootContainer />, document.elementById('react-root'));
+```
+you would add the following code to accept changes to RootContainer _or any of it's descendants_.
+
+```js
+ if (module.hot) {
+   module.hot.accept('./containers/rootContainer.js', () => {
+     const NextRootContainer = require('./containers/rootContainer.js').default;
+     render(<NextRootContainer />, document.elementById('react-root'));
+   }
+ }
+```
+Note, with no further steps, this enough to hotload changes to React components, but state will not be preserved.  If you externalize all your state in a state store like Redux, this might be enough.
+
+#### Step 3/3: Adding React Hot Loader to preserve state
+
+The final step adds `react-hot-loader` to our project to preserve *component state* across hot loads.
+
+1.  Install the package:
 
-Start the server we configured earlier via `npm start` and open the dev server URL in browser. To test hot reloading, just edit any component and watch the changes happen live!
+    ```sh
+    $ npm install --save-dev react-hot-loader
+    ```
+1.  Add the package to your config.
+
+    a.  If you use Babel, modify your `.babelrc` to ensure it includes at least:
+
+    ```js
+      {
+        "plugins": [ "react-hot-loader/babel" ]
+      }
+    ```
+    b. Alternatively, in Webpack, add `react-hot-loader/webpack` to the **loaders** section of your `webpackConfig.js`:
+
+    ```js
+        module: {
+            loaders: [{
+                test: /\.js$/,
+                loaders: ['react-hot-loader/webpack', 'babel'],
+                include: path.join(__dirname, 'src')
+            }]
+        }
+    ```
+
+    NB: `react-hot-loader/webpack` only works on *exported* components,
+    whereas `react-hot-loader/babel` picks up all *top-level variables* in
+    your files.
+
+    As a workaround, with webpack, you can export all the
+    components whose state you want to maintain, even if they're not
+    imported anywhere else.
+
+1.  Add following line to the top of your main entry point:
+
+    ```js
+    import 'react-hot-loader/patch';
+    ```
+
+1.  Wrap your `<RootContainer/>` inside of an `<AppContainer>`:
+
+    ```js
+    import { AppContainer } from 'react-hot-loader';
+    import RootContainer from './containers/rootContainer.js';
+
+    render((
+        <AppContainer>
+            <RootContainer />
+        </AppContainer>
+    ), document.getElementById('react-root'));
+    ```  
+
+    You should do this for both instances, e.g. your original mount and your mount code inside of the `module.hot.accept()` function.  `<AppContainer>` must wrap only a single, React component.
+
+That's it!
+
+### Putting it all together
+
+If you've gotten this far - you're almost done! But before showing you what your app's
```

Comment:
Agree.  I like the complete example at the end but yes, some of the explanations (thanks @tsaiDavid) there would be better suited in each relevant section.  Feel free to move it if you're already working on this otherwise one of us will do it later.